cmd/thanos/query.go

```diff
@@ -412,9 +415,38 @@ func runQuery(
 			router = router.WithPrefix(webRoutePrefix)
 		}
 
+		buildInfo := &v1.ThanosVersion{
+			Version:   version.Version,
+			Revision:  version.Revision,
+			Branch:    version.Branch,
+			BuildUser: version.BuildUser,
+			BuildDate: version.BuildDate,
+			GoVersion: version.GoVersion,
+		}
+
+		CWD, err := os.Getwd()
+		if err != nil {
+			CWD = "<error retrieving current working directory>"
+			level.Warn(logger).Log("msg", "failed to retrieve current working directory", "err", err)
+		}
+
+		birth := time.Now()
+
+		var runtimeInfo v1.RuntimeInfoFn = func() v1.RuntimeInfo {
+			status := v1.RuntimeInfo{
+				StartTime:      birth,
+				CWD:            CWD,
+				GoroutineCount: runtime.NumGoroutine(),
+				GOMAXPROCS:     runtime.GOMAXPROCS(0),
+				GOGC:           os.Getenv("GOGC"),
+				GODEBUG:        os.Getenv("GODEBUG"),
+			}
+			return status
```

Comment:
We could just return this struct directly, no need for a variable. But I'd guess that the compiler already does this optimization for us. Plus, I'm surprised that none of our linters caught this. Perhaps we could improve the linters to catch functions that look like:
```go
func foo() test {
  a := test{}
  return a
}
```

In the distant future? :smile: